### PR TITLE
feat: offline mode

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/hooks/useConnectionChange.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useState } from "react";
```
Comment (itsabdessalam) [inline src/hooks/useConnectionChange.jsx]: `useState` does not seem to be used in the entire file?

File: src/components/Offline.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { useConnectionChange } from "../hooks";
+import { useEffect, useState } from "react";
+
+const Offline = () => {
+  const [isDisconnected, setIsDisconnected] = useState(true);
+
+  useEffect(() => {
+    setIsDisconnected(useConnectionChange());
+    window.addEventListener("online", useConnectionChange());
+    window.addEventListener("offline", useConnectionChange());
```
Comment (itsabdessalam) [inline src/components/Offline.jsx]: You can use a cleanup function for `useEffect` to removeEventListener for both listeners

File: src/hooks/useConnectionChange.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+const useConnectionChange = () => {
+  let isDisconnected = false;
+  const condition = navigator.onLine ? "online" : "offline";
```
Comment (itsabdessalam) [inline src/hooks/useConnectionChange.jsx]: `status` is appropriate or why not just use `navigator.onLine` value?

File: src/hooks/useConnectionChange.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+          return clearInterval(webPing);
+        })
+        .catch(() => {
+          return (isDisconnected = true);
+        });
+    }, 2000);
+    return;
+  }
+
+  return isDisconnected;
```
Comment (itsabdessalam) [inline src/hooks/useConnectionChange.jsx]: It would be great to always return a value?

File: src/hooks/useConnectionChange.jsx (hunk 1 above)
Comment (ghost) [reply src/hooks/useConnectionChange.jsx]: I don't understand

File: src/hooks/useConnectionChange.jsx (hunk 3 above)
Comment (ghost) [reply src/hooks/useConnectionChange.jsx]: Yes maybe

File: src/hooks/useConnectionChange.jsx (hunk 4 above)
Comment (ghost) [reply src/hooks/useConnectionChange.jsx]: Ok by default make this at false

File: src/components/Offline.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { useConnectionChange } from "../hooks";
+import { useEffect, useState } from "react";
+
+const Offline = () => {
+  const [isDisconnected, setIsDisconnected] = useState(true);
+  useEffect(() => {
+    setIsDisconnected(useConnectionChange());
+    window.addEventListener("offline", useConnectionChange());
+    window.removeEventListener("offline", useConnectionChange());
```
Comment (itsabdessalam) [inline src/components/Offline.jsx]:
```
  return () => {
    // your cleanup code
  };
```

**Review by itsabdessalam — CHANGES_REQUESTED** [review]
https://github.com/itsabdessalam/taskee/pull/15/files#r534675936

**Review by itsabdessalam — APPROVED** [review]

**Pull request merged**